src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,83 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static class Pair<K, V> {
+        private K key;
+        private V value;
+
+        public Pair(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+
+        public K getKey() {
+            return key;
+        }
+
+        public void setKey(K key) {
+            this.key = key;
+        }
+
+        public V getValue() {
+            return value;
+        }
+
+        public void setValue(V value) {
+            this.value = value;
+        }
+    }
+
+    private static final int EMPTY_INDEX = -1;
+    private static final int MAX_ITEMS_NUMBER = 10;
+    private final Pair<K, V>[] arrOfPair;
+    private int size;
+
+    public StorageImpl() {
+        arrOfPair = new Pair[MAX_ITEMS_NUMBER];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (getElementIndex(key) == EMPTY_INDEX) {
+            arrOfPair[size] = getPair(key, value);
+            size++;
+        } else {
+            arrOfPair[getElementIndex(key)].setValue(getPair(key, value).getValue());
+        }
+    }
+
+    public Pair<K, V> getPair(K key, V value) {
+        Pair<K, V> pair = new Pair<>(key, value);
+        return pair;
     }
 
     @Override
     public V get(K key) {
-        return null;
+        V result = null;
+        for (int i = 0; i < size; i++) {
+            if ((arrOfPair[i].getKey() == key)
+                    || (arrOfPair[i].getKey() != null && arrOfPair[i].getKey().equals(key))) {
+                result = arrOfPair[i].getValue();
+                break;
```

Comment:
using **break** or **continue** is a bad practice
```suggestion
                return  arrOfPair[i].getValue();
```